deploy-service/common/src/main/java/com/pinterest/deployservice/db/DBPindeployDAOImpl.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2024 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *  
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *    
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.deployservice.db;
+
+import com.pinterest.deployservice.bean.PindeployBean;
+import com.pinterest.deployservice.bean.SetClause;
+import com.pinterest.deployservice.dao.PindeployDAO;
+import org.apache.commons.dbcp.BasicDataSource;
+import org.apache.commons.dbutils.QueryRunner;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.apache.commons.dbutils.handlers.BeanHandler;
+
+/* count table cache to store # of actively deploying agents and # of existing agents */
+public class DBPindeployDAOImpl implements PindeployDAO {
+    private static final String GET_PINDEPLOY =
+        "SELECT * FROM pindeploy WHERE env_id=?";
+    private static final String DELETE_PINDEPLOY =
+        "DELETE * FROM pindeploy WHERE env_id=?";
+    private static final String INSERT_OR_UPDATE_PINDEPLOY =
+        "INSERT INTO pindeploy SET %s ON DUPLICATE KEY UPDATE %s";
```

Comment:
We had a recent security risk for a query using %s in pindeploy, fix was to use ?, can see the commit here https://github.com/pinternal/optimus/commit/af9794a2e6c8ad2ba2305a24b9f92df9bbe69b82

Reply:
That might not be relevant since it looks like the clause can't be set to just anything

Reply:
Never mind, I updated it to enhance security. ^^